virion.md

```diff
@@ -51,9 +58,62 @@ and SHOULD avoid using it as a substring in their fully-qualified paths
 To avoid ambiguation, the virion namespace root
 MUST have at least two parts separated by a `\`.
 
-`spec` is the version of the virion specification (i.e. this document).
-Do not change this field unless
-the library requires features in newer versions of the virion specification.
+A virion MUST only export the following items under the namespace root:
+
+- Classes
+- Enums
+- Interfaces
+- Traits
+
+Namespace functions, namespace constants and global variables MUST NOT be used.
+
+### Shared namespace root
+
+`shared-namespace-root` is similar to `namespace-root`,
+but items declared under the shared namespace root are not shaded.
+This is to allow (potentially different versions of) the same virion
+to be used in multiple plugins as separate shaded instances
+but still communicate correctly through the shared namespace.
+
+The shared namespace root MUST NOT intersect with the namespace root.
+Conventionally, developers CAN use the form `Shared\{NamespaceRoot}` as the shared namespace root.
+
+A virion MUST only export the following items under the shared namespace root:
+
+- Enums
+- Interfaces
+  - Constant definitions must be resolvable at compile time.
+- Classes with only public fields and public constants.
+  - Abstract and final classes are allowed
+  - Public static fields are allowed.
+  - Constant definitions must be resolvable at compile time.
+
+Items in the shared namespace MUST NOT reference any shaded items,
+including shaded items declared by transitive dependencies.
+The term "reference" here includes parameter types, return types,
+`extends`/`implements` target and attributes.
+
+Once a virion has been published,
+the defition of all items under the shared namespace root MUST NOT ever be changed.
+In particular:
+
+- Semantics-sensitive attributes and doc comments MUST NOT be changed.
```

Comment:
even doc comment mistakes can't be fixed?

Reply:
@dktapps "semantics-sensitive" refers to doc comments too. Such as `@noHandle` or something like that.

Reply:
Normal doc comment changes are fine since they don't affect runtime.

Reply:
@noHandle is defined by PM. What happens if PM changes this in the future?

Reply:
ok maybe it is not so strictly required if two versions can't run simultaneously due to incompatible major. but better have stronger requirement to start with.